lib/chip/service.rb

```diff
@@ -44,6 +55,26 @@ def token
       end
     end
 
+    def with_monitoring_and_error_handling(&)
+      with_monitoring(2, &)
+    rescue => e
+      handle_error(error: e)
```

Comment:
Instead of calling `handle_error`, we can just log an exception in sentry & throw exception like below               
```
Chip::ServiceException.new(
                key: "CHIP_#{error.status}",
                original_status: error.status,
                original_body: error.body
              )
```